### ISSUE TITLE
Search: invalid PropType noResultMessage

### ISSUE THREAD

**Issue by jptissot** [Semantic-Org/Semantic-UI-React#1664]
### Steps
When using the Search component, I want to be able to pass a react component as a noResultMessage to allow for localization / links. This works as expected (It displays fine) but i get a proptype warning in the console.
### Expected Result
No warning in console
### Actual Result
 Invalid prop `noResultsMessage` of type `object` supplied to `Search`, expected `string`.
### Version
0.68.3

### Testcase

```
<Search
  noResultsMessage={<FormattedMessage {...messages.NoSurveysFound} values={{ link: <Link to="/" onClick={() => clear()}>{formatMessage(messages.ViewAllSurveys)}</Link> }} />}
/>
```

**Comment by jptissot** [comment]
Just a side note: I did try this 

``` JavaScript
<Search
  noResultsMessage={formatMessage(messages.NoSurveysFound, {link: <Link to="/" onClick={() => clear()}>{formatMessage(messages.ViewAllSurveys)}</Link> })}
/>
```

Doing this makes the `<Link />` element display as `Object [object]` instead of displaying a link.

**Comment by levithomason** [comment]
I believe we made the same updates to the Dropdown.  We simply need to relax the prop type validation here.  PRs welcome.

**Comment by levithomason** [comment]
Fixed in #1683

**Comment by lazazivkovic** [comment]
> I believe we made the same updates to the Dropdown. We simply need to relax the prop type validation here. PRs welcome.

@levithomason Could you please take a look at #3177 ? Thanks 😃
